File: python/数组/1030. 距离顺序排列矩阵单元格.py
from typing import List
class Solution:
    def allCellsDistOrder(self, R: int, C: int, r0: int, c0: int) -> List[List[int]]:
        from collections import  defaultdict
        d = defaultdict(list)
        for i in range(R):
            for j in range(C):
                d[abs(r0-i)+abs(c0-j)].append([i,j])
        res = []
        for i in sorted(d):
            res.extend(d[i])
        return res
s = Solution()
R = 2
C = 3
r0 = 1
c0 = 2
print(s.allCellsDistOrder(R,C,r0,c0))

File: go/数组/1030. 距离顺序排列矩阵单元格.go
package main

import "sort"

func allCellsDistOrder(R int, C int, r0 int, c0 int) [][]int {
	var res [][]int
	for i := 0; i < R; i++ {
		for j := 0; j < C; j++ {
			res = append(res, []int{i, j})
		}
	}
	sort.Slice(res, func(i, j int) bool {
		a, b := res[i], res[j]
		return abs(a[0]-r0)+abs(a[1]-c0) < abs(b[0]-r0)+abs(b[1]-c0)
	})
	return res
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
